exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/Query.kt

```diff
@@ -174,6 +218,9 @@ open class Query(override var set: FieldSet, where: Op<Boolean>?) : AbstractQuer
                 if (distinct) {
                     append("DISTINCT ")
                 }
+                distinctOn?.let { columns ->
+                    columns.appendTo(prefix = "DISTINCT ON (", postfix = ") ") { +"${it.table.tableName}.${it.name}" }
+                }
```

Comment:
Unless there's a reason I'm missing, I'd suggest using the existing column identifier logic instead of this lambda:
```kt
columns.appendTo(prefix = "DISTINCT ON (", postfix = ") ") { append(it) }
```
`append(it)` calls `Column.toQueryBuilder()`, which then calls `fullIdentity()` and gives the same result, but respects the database identifier case rules. The difference is only noticeable in H2 (which seems to be loose enough to not actually care), but it's probably more uniformed to apply consistent casing rules in every place that we can.